tests/Models/CharacterModelTest.php

```diff
@@ -104,4 +108,30 @@ public function testDeletion() {
         
         $this->assertEquals($rowsBefore - 1, $rowsAfter);
     }
+    
+    /**
+     * Tests character properties
+     */
+    public function testProperties()
+    {
+        $em = $this->getEntityManager();
+        
+        // test default values
+        $firstCharacter = $em->getRepository(Character::class)->find(1);
+        $this->assertSame(5, $firstCharacter->getProperty("dragonkills", 5));
+        $this->assertNotSame(5, $firstCharacter->getProperty("dragonkills", "5"));
+        $this->assertSame("hanniball", $firstCharacter->getProperty("petname", "hanniball"));
+        
+        // test setting variables, then getting
+        $firstCharacter->setProperty("dragonkills", 5);
+        $this->assertSame(5, $firstCharacter->getProperty("dragonkills"));
+        $this->assertNotSame("5", $firstCharacter->getProperty("dragonkills"));
+        
```

Comment:
This is good but it doesnt test writing it to the database and reading it back, correct? Just writing and reading from the in-memory properties instance variable.